apps/playground/src/components/playground/chat-ui.tsx

```
@@ -226,7 +227,25 @@ const AssistantMessage = memo(
 	}) => {
 		// useMemo for extracted parts to avoid recomputation
 		const { textParts, imageParts, toolParts, reasoningContent, sourceParts } =
-			useMemo(() => extractMessageParts(message.parts), [message.parts]);
+			useMemo(() => {
+				const result = extractMessageParts(message.parts);
+				// Debug: Log when tool parts are found
+				if (result.toolParts.length > 0) {
+					console.log("=== Tool parts found in message ===");
```

ESLint

[error] Unexpected console statement.

(no-console)